src/reformat/libyuv.rs

```diff
@@ -130,243 +130,202 @@ impl ConversionFunction {
     }
 }
 
-#[allow(clippy::single_match)]
 fn find_conversion_function(
     yuv_format: PixelFormat,
     yuv_depth: u8,
     rgb: &rgb::Image,
     alpha_preferred: bool,
 ) -> Option<ConversionFunction> {
-    if yuv_depth > 8 {
-        if yuv_format != PixelFormat::Yuv444 {
-            if alpha_preferred {
-                match yuv_depth {
-                    10 => match rgb.format {
-                        Format::Rgba | Format::Bgra => match yuv_format {
-                            PixelFormat::Yuv422 => {
-                                return Some(ConversionFunction::YUVAToRGBMatrixFilterHighBitDepth(
-                                    I210AlphaToARGBMatrixFilter,
-                                ))
-                            }
-                            PixelFormat::Yuv420 => {
-                                return Some(ConversionFunction::YUVAToRGBMatrixFilterHighBitDepth(
-                                    I010AlphaToARGBMatrixFilter,
-                                ))
-                            }
-                            _ => {}
-                        },
-                        _ => {}
-                    },
-                    _ => {}
-                }
-            }
-            match yuv_depth {
-                10 => match rgb.format {
-                    Format::Rgba | Format::Bgra => match yuv_format {
-                        PixelFormat::Yuv422 => {
-                            return Some(ConversionFunction::YUVToRGBMatrixFilterHighBitDepth(
-                                I210ToARGBMatrixFilter,
-                            ))
-                        }
-                        PixelFormat::Yuv420 => {
-                            return Some(ConversionFunction::YUVToRGBMatrixFilterHighBitDepth(
-                                I010ToARGBMatrixFilter,
-                            ))
-                        }
-                        _ => {}
-                    },
-                    _ => {}
-                },
-                _ => {}
-            }
+    match (alpha_preferred, yuv_depth, rgb.format, yuv_format) {
+        (true, 10, Format::Rgba | Format::Bgra, PixelFormat::Yuv422)
```

Comment:
almost most all of these cases have `Format::Rgba | Format::Bgra` except a few. so it might avoid some repetition if you make it a two level match instead of a flattened single level match?

```
match rgb.format {
  Rgba | Bgra => match (alpha_preferred, yuv_depth, yuv_format) { }.
  Rgb | Bgr => match (alpha_preferred, yuv_depth, yuv_format) { }.
  .. and so on.
}
```

Reply:
I like the single-level match for two reasons:
- The full context is accessible right above the result, no need to remember higher conditions or to scroll up to look it up,
- It checks for unreachable branches. I do not know if it works with nested match blocks but at least it was not detected before this PR.

Reply:
Ok makes sense.